src/Subscriber/PaymentSubscriber.php

```diff
@@ -452,4 +425,46 @@ public function onKernelRequest(RequestEvent $event)
             );
         }
     }
+
+    private function getGiftcardTotalDiscount($salesChannelContext, $amountInMinorUnits)
+    {
+        $data = $this->paymentStateDataService
+            ->fetchRedeemedGiftCardsFromContextToken($salesChannelContext->getToken());
+
+        $totalGiftcardBalance = 0;
+
+        foreach ($data->getElements() as $statedataArray) {
+            $stateData = json_decode($statedataArray->getStateData(), true);
+            if (isset($stateData['paymentMethod']['type']) ||
+                isset($stateData['paymentMethod']['brand']) ||
+                $stateData['paymentMethod']['type'] === 'giftcard') {
+                $totalGiftcardBalance += $stateData['giftcard']['value'];
+            }
+        }
+
+        if ($totalGiftcardBalance > 0) {
+            $totalDiscount =  min($totalGiftcardBalance, $amountInMinorUnits);
+        } else {
+            $totalDiscount = 0;
+        }
+
+        return [
+            'giftcardDiscount' => $totalDiscount,
+            'giftcardBalance' => $totalGiftcardBalance,
```

Comment:
I didn't see any usage for `giftcardBalance` left on the frontend. Why not just return `giftcardDiscount`?

Reply:
Hey, We do use it on onShoppingCartLoaded()